Repository/PhpcrRepository.php

```diff
@@ -119,4 +125,85 @@ protected function buildCollection(array $nodes)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        try {
+            $parentNode = $this->session->getNode($resolvedPath);
+        } catch (PathNotFoundException $e) {
+            throw new InvalidArgumentException('No parent node created for '.$path, $e->getCode(), $e);
+        }
+
+        /** @var PhpcrResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::notNull($resource->getName(), 'The resource needs a name for the creation.');
+            Assert::notNull($resource->getPayloadType(), 'The resource needs a type for the creation');
+
+            $parentNode->addNode($resource->getName(), $resource->getPayloadType());
+        }
+
+        $this->session->save();
+    }
+
+    /**
+     * Moves a resource inside the repository.
+     *
+     * @param string $sourceQuery The Path of the current document.
+     * @param string $targetPath  The parent path of the destination.
+     * @param string $language
+     *
+     * @return int
+     */
+    public function move($sourceQuery, $targetPath, $language = 'glob')
+    {
+        $this->failUnlessGlob($language);
+        Assert::notEq('', trim($sourceQuery, '/'), 'The root directory cannot be moved.');
+
+        $targetPath = $this->resolvePath($targetPath);
+        $sourcePath = $this->resolvePath($sourceQuery);
+
+        try {
+            $this->session->move($sourcePath, $targetPath);
+
+            return 1;
+        } catch (PathNotFoundException $e) {
+            throw new \InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        throw new \Exception('Clear currently not supported');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function removeResource($sourcePath, $deleted)
+    {
+        $deleted += count($this->session->getNode($sourcePath)->getNodes()) + 1;
+
+        try {
+            $this->session->removeItem($sourcePath);
+        } catch (PathNotFoundException $e) {
+            throw new \InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
```

Comment:
PhpcrRepositoryTest::testRemoveThrowsWhenSessionThrows()
Same here: i will add the message assertion.